packages/backend-for-frontend/src/services/validators.ts

```diff
@@ -208,3 +210,13 @@ export function assertIsDraftEservice(eservice: catalogApi.EService): void {
     throw eserviceIsNotDraft(eservice.id);
   }
 }
+
+export function assertEServiceIsTemplateInstance(
+  eservice: catalogApi.EService
+): asserts eservice is catalogApi.EService & {
+  templateRef: NonNullable<EService["templateRef"]>;
+} {
+  if (eservice.templateRef !== undefined) {
+    throw eserviceIsTemplateInstance(eservice.id, eservice.templateRef.id);
```

Comment:
Could you copy this assertion, error, naming etc., from how we do it in the catalog-process?
So that we don't  do the same thing in two different ways
https://vscode.dev/github.com/pagopa/interop-be-monorepo/blob/develop/packages/catalog-process/src/services/validators.ts#L276